source/local/local.go

```diff
@@ -144,14 +160,54 @@ func (s *localSource) GetFeatures() *nfdv1alpha1.Features {
 	return s.features
 }
 
-func parseFeatures(lines [][]byte) map[string]string {
+func parseDirectives(line string, opts *parsingOpts) error {
+	lineSplit := strings.SplitN(line, "=", 2)
+
+	// If we have the following directive: # +expiry-time=xxx
+	// we need to get the key '# +expiry-time' and then we need to extract
+	// the following key 'expiry-time'.
+	key := strings.Split(lineSplit[0], DirectivePrefix)[1]
+	if len(lineSplit) == 1 {
+		return ErrorInvalidDirectiveFormat
+	}
+	value := lineSplit[1]
+
+	switch key {
+	case ExpiryTimeKey:
+		expiryDate, err := time.Parse(time.RFC3339, strings.TrimSpace(value))
+		if err != nil {
+			return fmt.Errorf("failed to parse expiry-date directive: %w", err)
+		}
+		opts.ExpiryTime = expiryDate
+	default:
+		return fmt.Errorf("unknown feature file directive %q", key)
+	}
+
+	return nil
+}
+
+func parseFeatures(lines [][]byte, fileName string) map[string]string {
 	features := make(map[string]string)
+	now := time.Now()
+	parsingOpts := &parsingOpts{
+		ExpiryTime: now,
+	}
 
 	for _, l := range lines {
 		line := strings.TrimSpace(string(l))
 		if len(line) > 0 {
-			// Skip comment lines
-			if strings.HasPrefix(line, "#") {
+			if strings.HasPrefix(line, DirectivePrefix) {
```

Comment:
We should still allow "normal" comment lines (like `# Note: foo bar`) without trying to parse these as features

```suggestion
			if strings.HasPrefix(line, "#") {
```